controllers/drplacementcontrol_controller.go

```diff
@@ -1602,65 +1742,224 @@ func (s ObjectStorePVDownloader) DownloadPVs(ctx context.Context, r client.Reade
+		}
+	}
+
+	d.log.Info(fmt.Sprintf("Updated Conditions '%v'", d.instance.Status.Conditions))
 }
 
-func (d *DRPCInstance) setDRState(nextState rmn.DRState) {
-	if d.instance.Status.LastKnownDRState != nextState {
-		d.log.Info(fmt.Sprintf("LastKnownDRState: Current State '%s'. Next State '%s'",
-			d.instance.Status.LastKnownDRState, nextState))
+func (d *DRPCInstance) newCondition(condType string) *metav1.Condition {
+	return &metav1.Condition{
+		Type:   condType,
+		Status: d.getConditionStatus(condType),
+		// ObservedGeneration: d.getObservedGeneration(condType),
+		LastTransitionTime: metav1.Now(),
+		Reason:             d.getConditionReason(condType),
+		Message:            d.getConditionMessage(condType),
+	}
+}
 
-		d.instance.Status.LastKnownDRState = nextState
-		d.needStatusUpdate = true
+// func (d *DRPCInstance) getConditionType() string {
+// 	if d.instance.Status.Phase == "" {
+// 		return rmn.ConditionNotAvailable
+// 	}
+
+// 	return rmn.ConditionAvailable
+// }
+
+func (d *DRPCInstance) getConditionStatus(condType string) metav1.ConditionStatus {
+	if condType == rmn.ConditionAvailable {
+		return d.getConditionStatusForTypeAvailable()
+	} else if condType == rmn.ConditionReconciling {
+		return d.getConditionStatusForTypeReconciling()
+	}
+
+	return metav1.ConditionUnknown
+}
+
+func (d *DRPCInstance) getConditionStatusForTypeAvailable() metav1.ConditionStatus {
+	if d.isInFinalPhase() {
+		return metav1.ConditionTrue
+	}
+
+	if d.isInProgressingPhase() {
+		return metav1.ConditionFalse
 	}
+
+	return metav1.ConditionUnknown
 }
 
+func (d *DRPCInstance) getConditionStatusForTypeReconciling() metav1.ConditionStatus {
+	if d.isInFinalPhase() {
+		if d.reconciling() {
+			return metav1.ConditionFalse
+		}
+
+		return metav1.ConditionTrue
+	}
+
+	if d.isInProgressingPhase() {
+		return metav1.ConditionFalse
+	}
+
+	return metav1.ConditionUnknown
+}
+
+func (d *DRPCInstance) reconciling() bool {
+	idx, condition := GetDRPCCondition(&d.instance.Status, rmn.ConditionReconciling)
+	d.log.Info(fmt.Sprintf("Condition %v", condition))
+
+	if idx == -1 {
+		return false
+	}
+
+	if condition.Reason == rmn.ReasonSuccess && condition.Status == metav1.ConditionTrue {
+		return false
+	}
+
+	return true
+}
+
+func (d *DRPCInstance) getConditionReason(condType string) string {
+	if condType == rmn.ConditionAvailable {
+		return string(d.instance.Status.Phase)
+	} else if condType == rmn.ConditionReconciling {
+		return d.getReasonForConditionTypeReconciling()
+	}
+
+	return rmn.ReasonUnknown
+}
+
+func (d *DRPCInstance) getReasonForConditionTypeReconciling() string {
+	if d.isInFinalPhase() {
+		if d.reconciling() {
+			return rmn.ReasonCleaning
+		}
+
+		return rmn.ReasonSuccess
+	}
+
+	if d.isInProgressingPhase() {
+		return rmn.ReasonProgressing
+	}
+
+	return rmn.ReasonUnknown
+}
+
+func (d *DRPCInstance) getConditionMessage(condType string) string {
+	return fmt.Sprintf("Condition type %s", condType)
```

Comment:
If message is optional we can drop this addition, till we get better messages from points in the code where the conditions are initially set.